File: src/main/java/com/javarush/island/shevchenko/entities/animals/herbivores/Rabbit.java
package com.javarush.island.shevchenko.entities.animals.herbivores;

import com.javarush.island.shevchenko.entities.organism.Animal;
import com.javarush.island.shevchenko.entities.organism.interfaces.*;
import com.javarush.island.shevchenko.settings.IslandSettings;
import com.javarush.island.shevchenko.util.Randomizer;

/**
 * Класс Rabbit представляет существо кролик на острове животных.
 * Кролики являются травоядными и используют поведение из интерфейса Herbivorous для питания.
 */
public class Rabbit extends Animal implements Herbivorous {

    public Rabbit() {
        this.aClass = Rabbit.class;
        this.weight = Randomizer.getRandom(IslandSettings.ANIMAL_PARAMETERS.get(Rabbit.class)[0] / 2, IslandSettings.ANIMAL_PARAMETERS.get(Rabbit.class)[0]);
    }
}

File: src/main/kotlin/ru/cooper/island/entities/animals/herbivores/Rabbit.kt
package ru.cooper.island.entities.animals.herbivores

import ru.cooper.island.entities.animals.herbivores.function.initAnimal
import ru.cooper.island.entities.organism.Animal
import ru.cooper.island.entities.organism.interfaces.Herbivorous

class Rabbit : Animal(), Herbivorous {
    init { initAnimal<Rabbit>() }
}
